source/Body.cpp

```diff
@@ -205,90 +251,399 @@ const Government *Body::GetGovernment() const
+			spriteAnimationParameters.transitionDelay = child.Value(1);
 		else if(child.Token(0) == "scale" && child.Size() >= 2 && child.Value(1) > 0.)
-			scale = static_cast<float>(child.Value(1));
+			spriteAnimationParameters.scale = static_cast<float>(child.Value(1));
 		else if(child.Token(0) == "start frame" && child.Size() >= 2)
 		{
-			frameOffset += static_cast<float>(child.Value(1));
-			startAtZero = true;
+			spriteAnimationParameters.startFrame = static_cast<float>(child.Value(1));
+			spriteAnimationParameters.startAtZero = true;
+		}
+		else if(child.Token(0) == "ramp" && child.Size() >= 2)
+		{
+			if(child.Size() == 2)
+			{
+				spriteAnimationParameters.rampUpRate = child.Value(1) / 3600.;
+				spriteAnimationParameters.rampDownRate = child.Value(1) / 3600.;
+			}
+			else if(child.Size() == 3)
+			{
+				spriteAnimationParameters.rampUpRate = child.Value(1) / 3600.;
+				spriteAnimationParameters.rampDownRate = child.Value(2) / 3600.;
+			}
+		}
+		else if(child.Token(0) == "indicate percentage" && child.Size() >= 2 && child.Value(1) > 0. && child.Value(1) < 1.)
+		{
+			spriteAnimationParameters.indicateReady = true;
+			spriteAnimationParameters.repeat = false;
+			spriteAnimationParameters.startAtZero = true;
+			spriteAnimationParameters.indicatePercentage = static_cast<float>(child.Value(1));
+		}
+		else if(child.Token(0) == "no indicate")
+			spriteAnimationParameters.indicateReady = false;
+		else if(child.Token(0) == "indicate")
+		{
+			spriteAnimationParameters.indicateReady = true;
+			spriteAnimationParameters.repeat = false;
+			spriteAnimationParameters.startAtZero = true;
+			spriteAnimationParameters.indicatePercentage = 1.0f;
 		}
+		else if(child.Token(0) == "random")
+			spriteAnimationParameters.randomize = true;
 		else if(child.Token(0) == "random start frame")
-			randomize = true;
+			spriteAnimationParameters.randomizeStart = true;
 		else if(child.Token(0) == "no repeat")
 		{
-			repeat = false;
-			startAtZero = true;
+			spriteAnimationParameters.repeat = false;
+			spriteAnimationParameters.startAtZero = true;
+		}
+		else if(child.Token(0) == "transition rewind")
+		{
+			spriteAnimationParameters.transitionFinish = false;
+			spriteAnimationParameters.transitionRewind = true;
+		}
+		else if(child.Token(0) == "transition finish")
+		{
+			spriteAnimationParameters.transitionFinish = true;
+			spriteAnimationParameters.transitionRewind = false;
 		}
 		else if(child.Token(0) == "rewind")
-			rewind = true;
-		else if(child.Token(0) == "center" && child.Size() >= 3)
-			center = Point(child.Value(1), child.Value(2));
+			spriteAnimationParameters.rewind = true;
+		else if(child.Token(0) == "reverse")
+			spriteAnimationParameters.reverse = true;
 		else
 			child.PrintTrace("Skipping unrecognized attribute:");
 	}
 
+	// Now load the trigger sprites.
+	for(int val = 0; val < static_cast<int>(triggerSpriteDefer.size()); ++val)
+	{
+		DataNode node = triggerSpriteDefer.at(val);
```

Comment:
You don't need bounds checking here as the loop is limited to the size of the vector.
```suggestion
		DataNode node = triggerSpriteDefer[val];
```